rust/core-lib/src/view.rs

```diff
@@ -242,6 +243,55 @@ impl View {
         self.set_selection_raw(text, selection);
     }
 
+    /// Selects an entire word and supports multi selection.
+    pub fn select_word(&mut self, text: &Rope, offset: usize, multi_select: bool) {
+        let (start, end) = {
+            let mut word_cursor = WordCursor::new(text, offset);
+            word_cursor.select_word()
+        };
+
+        let region = SelRegion {
+            start: start,
+            end: end,
+            horiz: None,
+            affinity: Affinity::default(),
+        };
+
+        let mut selection = Selection::new();
+
+        if multi_select {
+            selection = self.selection.clone();
+        }
+
+        selection.add_region(region);
+        self.set_selection(text, selection);
+
+        self.start_drag(offset, start, end);
```

Comment:
So basically, 260-269 is the code to create the multi-selection.

Where is the `multi_select` flag set?

What would be a better option?

1. Create a function for this (which is repeated in the other function) and and add it to the original `editor.rs`. Or
2. Leave it as it is.
3. Create a function for the repeated identical part?

I'm wondering here...

Reply:
@rscircus your intuition is correct, the client sends a different message to core depending on whether or not ctrl/cmd is present during the click.

@scholtzan I do agree that there's possibly room for some code reuse, but I think this is fine as-is. Most of the ways this could be cleaner would require some api additions around `Selection` and `SelRegion`.